service-base/src/main/java/org/eclipse/hono/service/tenant/BaseTenantService.java

```diff
@@ -94,53 +109,113 @@ protected final String getEventBusAddress() {
+     */
+    private Future<EventBusMessage> processGetByCaRequest(final EventBusMessage request, final String subjectDn,
+            final Span span) {
 
         try {
             final X500Principal dn = new X500Principal(subjectDn);
             log.debug("retrieving tenant [subject DN: {}]", subjectDn);
             final Future<TenantResult<JsonObject>> getResult = Future.future();
-            get(dn, request.getSpanContext(), getResult.completer());
+            get(dn, span, getResult.completer());
             return getResult.map(tr -> {
                 final EventBusMessage response = request.getResponse(tr.getStatus())
                         .setJsonPayload(tr.getPayload())
                         .setCacheDirective(tr.getCacheDirective());
                 if (tr.isOk() && tr.getPayload() != null) {
-                    response.setTenant(getTypesafeValueForField(String.class, tr.getPayload(),
-                            TenantConstants.FIELD_PAYLOAD_TENANT_ID));
+                    final String tenantId = getTypesafeValueForField(String.class, tr.getPayload(),
+                            TenantConstants.FIELD_PAYLOAD_TENANT_ID);
+                    span.setTag(MessageHelper.APP_PROPERTY_TENANT_ID, tenantId);
+                    response.setTenant(tenantId);
                 }
+                Tags.HTTP_STATUS.set(span, tr.getStatus());
+                if (tr.isError()) {
+                    Tags.ERROR.set(span, true);
+                }
+                span.finish();
                 return response;
+            }).recover(t -> {
+                TracingHelper.logError(span, t);
+                span.finish();
+                return Future.failedFuture(t);
             });
         } catch (final IllegalArgumentException e) {
+            TracingHelper.logError(span, "illegal subject DN provided by client: " + subjectDn);
+            span.finish();
             // the given subject DN is invalid
             log.debug("cannot parse subject DN [{}] provided by client", subjectDn);
             return Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST));
         }
     }
 
+    /**
+     * Creates a new <em>OpenTracing</em> span for tracing the execution of a tenant service operation.
+     * <p>
+     * The returned span will already contain a tag for the given tenant (if it is not {code null}).
+     *
+     * @param operationName The operation name that the span should be created for.
+     * @param spanContext Existing span context.
+     * @param tenantId The tenant id.
+     * @return The new {@code Span}.
+     */
+    protected final Span newChildSpan(final String operationName, final SpanContext spanContext, final String tenantId) {
+        // we set the component tag to the class name because we have no access to
+        // the name of the enclosing component we are running in
+        final Tracer.SpanBuilder spanBuilder = tracer.buildSpan(operationName)
```

Comment:
what if operation is `null`?

Reply:
I've added a null check and documented the NPE thrown in that case.